category/functor.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2017 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Standard identity and composition functors
+-/
+import tactic.ext
+import category.basic
+
+universe variables u v w u' v' w'
+
+section functor
+
+variables {F : Type u → Type v}
+variables {α β γ : Type u}
+variables [functor F] [is_lawful_functor F]
+
+lemma functor.map_id : functor.map id = (id : F α → F α) :=
```

Comment:
remove this, you only apply it once in a context where you actually can just use `funext`

Reply:
Are you suggesting I rewrite the proof that relies on it this way?

```lean
protected lemma id_map : ∀ (x : comp f g α), comp.map id x = x
| ⟨x⟩ :=
by { have : map id = id := funext (@is_lawful_functor.id_map g _ _ α),
     simp [comp.map,this] }
```

Reply:
Also, those lemmas (with `functor.map_comp_map`) are useful in proofs with `traversable` (coming soon in a theatre near you),

Reply:
Are they so expensive to keep?

Reply:
I would prefer not having any theorems about `functor.map` when `<$>` will do. If you need this theorem, you can restate it as `(<$>) id = id`.